data/virt_autotest/clean_up_virt_logs.sh

```diff
@@ -0,0 +1,173 @@
+	done
+
+	if [[ ${extra_logs[@]} != "" ]];then
+	   for eachlog in ${extra_logs[@]};do
+	       echo -e "Going to clean up all empty and text log files in ${eachlog}." | tee -a ${clean_up_virt_logs_log}
+	       cleanup_logs ${eachlog}
+	       ret_result=$?
+	       echo -e "Cleaned up $? empty and text log files in ${eachlog}." | tee -a ${clean_up_virt_logs_log}
+	       total_cleanup=$(( ${total_cleanup} + ${ret_result} ))
+	   done
+	fi
+
+	if [[ ${total_cleanup} -gt 0 ]];then
+	   echo -e "Cleaned up ${total_cleanup} empty and text log files in total." | tee -a ${clean_up_virt_logs_log}
+	   return 0
+	else
+	   echo -e "Did not clean up any empty or text log files." | tee -a ${clean_up_virt_logs_log}
+	   return 1
+	fi
+}
+
+#Just power on or reboot active guests to give them a clear start
+function do_cleanup_on_guests() {
+	local guest_domain_types="sles"
+	local guests_inactive_array=`virsh list --inactive | grep -E "${guest_domain_types}" | awk '{print $2}'`
+	local guest_domains_array=`virsh list  --all | grep -E "${guest_domain_types}" | awk '{print $2}'`
+	local guest_current=""
+	local ret_result=0
+
+	for guest_current in ${guest_domains_array[@]};do
+	    if [[ ${guests_inactive_array[@]} =~ .*${guest_current}.* ]];then
+	       echo -e "Virtual machine ${guest_current} was inactive. Going to start it." | tee -a ${clean_up_virt_logs_log}
+	       echo -e "virsh start ${guest_current}"
+	       virsh start ${guest_current}
+	       ret_result=$(( ${ret_result} | $? ))
+	    else
+	       echo -e "Virtual machine ${guest_current} was running. Going to reboot it." | tee -a ${clean_up_virt_logs_log}
+	       echo -e "virsh reboot ${guest_current}"
+	       virsh reboot ${guest_current}
+	       ret_result=$(( ${ret_result} | $? ))
+	    fi
+	done
+
+	return ${ret_result}
+}
+
+#Usage and help info for the script
+help_usage(){
+	echo "script usage: $(basename $0) [-l \"Extra folders or files to be cleaned up on host,for example,\"log_file_1 log_file_2 log_folder_1\"(Can be omitted/Default to nothing)\"] [-h help]"
+}
+
+clean_up_virt_logs_log="/var/log/clean_up_virt_logs.log"
+virt_extra_logs_host=""
+clean_up_virt_logs_result=0
+rm -f -r ${clean_up_virt_logs_log}
+
+#Parse input arguments, all options are optional
+#Any log parameter passed in takes absolute path form
+while getopts 'l:h' OPTION; do
+   case "$OPTION" in
+      l)
+        virt_extra_logs_host="$OPTARG"
+        echo "The extra logs to be cleaned up are ${virt_extra_logs_host}" | tee -a ${fetch_logs_from_guest_log}
+        ;;
+      h)
+        help_usage | tee -a ${fetch_logs_from_guest_log}
+        exit 1
+        ;;
+      *)
+        help_usage | tee -a ${fetch_logs_from_guest_log}
+        exit 1
+        ;;
+   esac
+done
+shift "$(($OPTIND -1))"
+
+do_cleanup_on_host ${virt_extra_logs_host}
+clean_up_virt_logs_result=$(( ${clean_up_virt_logs_result} | $? ))
+do_cleanup_on_guests
+clean_up_virt_logs_result=$(( ${clean_up_virt_logs_result} | $? ))
+exit 0
```

Comment:
What is $clean_up_virt_logs_result used for if always exit 0?

Reply:
It records result, so keep records. But this script must exit 0 to avoid failure in test run. There is no point to trigger failure because of this.